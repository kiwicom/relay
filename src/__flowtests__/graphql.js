// @flow

import { graphql } from '../index';

module.exports = {
  correctUsage() {
    return graphql`correct usage`; // eslint-disable-line relay/graphql-syntax
  },
  correctUsageUnusual() {
    return graphql(['correct']);
  },
  incorrectUsage() {
    // $FlowExpectedError: input should be only string
    return graphql([1, 2]);
  },
};
