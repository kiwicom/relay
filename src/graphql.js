// @flow

import { graphql as _graphql } from 'react-relay';

import type { GraphQLTaggedNode } from './types.flow';

export default function graphql(
  strings: $ReadOnlyArray<string>,
): GraphQLTaggedNode {
  return _graphql(strings);
}
