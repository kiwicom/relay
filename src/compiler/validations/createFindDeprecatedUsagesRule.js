// @flow strict

import { findDeprecatedUsages, type DocumentNode, type ValidationContext } from 'graphql';
import { warning } from '@kiwicom/js';

export default function createFindDeprecatedUsagesRule(throwErrors: boolean = true) {
  return function findDeprecatedUsagesRule(validationContext: ValidationContext) {
    const schema = validationContext.getSchema();
    return {
      Document(node: DocumentNode): void {
        const deprecatedUsages = findDeprecatedUsages(schema, node);
        deprecatedUsages.forEach(error => {
          if (throwErrors) {
            throw error;
          } else {
            warning(false, '! %s', error.message);
          }
        });
      },
    };
  };
}
