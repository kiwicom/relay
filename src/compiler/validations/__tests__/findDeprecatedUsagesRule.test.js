// @flow

import path from 'path';
import { parse, validate } from 'graphql';
import { generateTestsFromFixtures } from '@kiwicom/test-utils';

import testSchema from './testSchema';
import createFindDeprecatedUsagesRule from '../createFindDeprecatedUsagesRule';

generateTestsFromFixtures(path.join(__dirname, 'fixtures', 'findDeprecatedUsagesRule'), query => {
  const documentAST = parse(query);
  const findDeprecatedUsagesRule = createFindDeprecatedUsagesRule(true);
  const errors = validate(testSchema, documentAST, [findDeprecatedUsagesRule]);
  return errors.length === 0 ? 'PASSED âœ…' : errors;
});
