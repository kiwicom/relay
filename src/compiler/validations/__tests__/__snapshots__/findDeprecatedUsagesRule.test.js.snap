// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches expected output: deprecatedAlias.graphql 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  alias: fieldDeprecated1
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
THROWN EXCEPTION:

The field QueryRoot.fieldDeprecated1 is deprecated. Use field "fieldString" instead.

GraphQL request:2:3
1 | {
2 |   alias: fieldDeprecated1
  |   ^
3 | }
`;

exports[`matches expected output: multipleDeprecated.graphql 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  fieldDeprecated1
  fieldDeprecated2
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
THROWN EXCEPTION:

The field QueryRoot.fieldDeprecated1 is deprecated. Use field "fieldString" instead.

GraphQL request:2:3
1 | {
2 |   fieldDeprecated1
  |   ^
3 |   fieldDeprecated2
`;

exports[`matches expected output: simple.graphql 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  fieldString
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
PASSED
`;

exports[`matches expected output: simpleWithDeprecated.graphql 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
{
  fieldString
  fieldDeprecated1
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
THROWN EXCEPTION:

The field QueryRoot.fieldDeprecated1 is deprecated. Use field "fieldString" instead.

GraphQL request:3:3
2 |   fieldString
3 |   fieldDeprecated1
  |   ^
4 | }
`;
